Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/ui/homecards/TestInvalidCard.kt

```diff
@@ -0,0 +1,25 @@
+package de.rki.coronawarnapp.submission.ui.homecards
+
+import android.view.ViewGroup
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.databinding.HomeSubmissionStatusCardInvalidBinding
+import de.rki.coronawarnapp.submission.ui.homecards.TestInvalidCard.Item
+import de.rki.coronawarnapp.ui.main.home.HomeAdapter
+
+class TestInvalidCard(parent: ViewGroup) : HomeAdapter.HomeItemVH<Item, HomeSubmissionStatusCardInvalidBinding>(
+    R.layout.home_card_container_layout, parent
+) {
+
+    override val viewBinding = lazy {
+        HomeSubmissionStatusCardInvalidBinding.inflate(layoutInflater, itemView.findViewById(R.id.card_container), true)
```

Comment:
This looks similar across all / most subclasses of `HomeAdapter.HomeItemVH`… I think it should be prettier to declare them in the superclass instead, but somehow that doesn't seem to be possible?

Reply:
Hm, only if we could call somehow pass the binding class and call that in the superclass with the necessary arguments, don't see it either, the inflation is via static methods, I'd not like to use reflection.

The base inflation into `itemView.findViewById(R.id.card_container)` is currently all the same, but there will be future features that have a home item that is not the default card.

So maybe we are okay with just leaving it as it is.

Reply:
Ah, yes, reflection should work, but that wouldn't be too much better either.

> there will be future features that have a home item that is not the default card

That wouldn't be an issue if it could simply be overwritten.

It also looks to me like there is no common superclass offering the `inflate` method. If view binding wasn't used, one would call `LayoutInflater.inflate`, and there one could unify that by having sublasses provide just the layout ID to their superclass, but that wouldn't work here either…

So yes, it's probably best left as is, though I'm not very happy with it personally.

Reply:
hmm, what about `DataBindingUtil.inflate`? we already pass the binding type as a generic argument

Reply:
@kolyaopahle That's a good idea as this class is made for that purpose and it looks to me like it would work. We'd need to pass the layout ID to the superclass so that it can inflate it, though.